// chain-registry/testnet/named.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.lumenxtestnet = exports.likecointestnet = exports.lavatestnet1 = exports.lavatestnet = exports.kyvetestnet = exports.kujiratestnet = exports.kichaintestnet = exports.junotestnet = exports.jackaltestnet = exports.injectivetestnet = exports.imversedtestnet = exports.impacthubtestnet = exports.hypersigntestnet = exports.humanstestnet = exports.hedgetestnet = exports.gitopiatestnet = exports.galacticatestnet = exports.finschiatestnet = exports.fetchhubtestnet = exports.evmostestnet = exports.entrypointtestnet = exports.empowertestnet = exports.elystestnet = exports.dydxtestnet = exports.doravotatestnet2 = exports.doravotatestnet = exports.desmostestnet = exports.deardogetestnet = exports.cudostestnet = exports.cosstestnet = exports.cosmwasmtestnet = exports.cosmoshubtestnet = exports.coreumtestnet = exports.coolcattestnet = exports.composabletestnet = exports.chimbatestnet = exports.cheqdtestnet = exports.chain4energytestnet = exports.celestiatestnet3 = exports.celestiatestnet = exports.cascadiatestnet = exports.bitcannadevnet = exports.babylontestnet1 = exports.babylontestnet = exports.axelartestnet = exports.auratestnet = exports.artelatestnet = exports.arkeonetworktestnet = exports.archwaytestnet = exports.akashtestnet = void 0;
exports.zetachaintestnet = exports.xiontestnet = exports.wavehashtestnet = exports.wardenprotocoltestnet = exports.vincechaintestnet = exports.upticktestnet = exports.uniontestnet = exports.ulastestnet = exports.titantestnet = exports.terra2testnet = exports.terpnettestnet2 = exports.terpnettestnet = exports.temporaltestnet = exports.swisstroniktestnet = exports.stridetestnet = exports.statesettestnet = exports.stargazetestnet = exports.sourcetestnet = exports.sixtestnet = exports.sgetestnet4 = exports.sgetestnet = exports.selfchaindevnet = exports.seitestnet2 = exports.seitestnet = exports.secretnetworktestnet2 = exports.secretnetworktestnet = exports.sagatestnet = exports.rsprovidertestnet = exports.qwoyntestnet = exports.quicksilvertestnet = exports.quasartestnet = exports.pryzmtestnet = exports.persistencetestnet2 = exports.persistencetestnet = exports.permtestnet = exports.osmosistestnet4 = exports.osmosistestnet = exports.okp4testnet = exports.nolustestnet = exports.noistestnet = exports.nobletestnet = exports.neutrontestnet = exports.migalootestnet = exports.marstestnet = exports.mantrachaintestnet = void 0;
exports.akashtestnet = __importStar(require("./akashtestnet"));
exports.archwaytestnet = __importStar(require("./archwaytestnet"));
exports.arkeonetworktestnet = __importStar(require("./arkeonetworktestnet"));
exports.artelatestnet = __importStar(require("./artelatestnet"));
exports.auratestnet = __importStar(require("./auratestnet"));
exports.axelartestnet = __importStar(require("./axelartestnet"));
exports.babylontestnet = __importStar(require("./babylontestnet"));
exports.babylontestnet1 = __importStar(require("./babylontestnet1"));
exports.bitcannadevnet = __importStar(require("./bitcannadevnet"));
exports.cascadiatestnet = __importStar(require("./cascadiatestnet"));
exports.celestiatestnet = __importStar(require("./celestiatestnet"));
exports.celestiatestnet3 = __importStar(require("./celestiatestnet3"));
exports.chain4energytestnet = __importStar(require("./chain4energytestnet"));
exports.cheqdtestnet = __importStar(require("./cheqdtestnet"));
exports.chimbatestnet = __importStar(require("./chimbatestnet"));
exports.composabletestnet = __importStar(require("./composabletestnet"));
exports.coolcattestnet = __importStar(require("./coolcattestnet"));
exports.coreumtestnet = __importStar(require("./coreumtestnet"));
exports.cosmoshubtestnet = __importStar(require("./cosmoshubtestnet"));
exports.cosmwasmtestnet = __importStar(require("./cosmwasmtestnet"));
exports.cosstestnet = __importStar(require("./cosstestnet"));
exports.cudostestnet = __importStar(require("./cudostestnet"));
exports.deardogetestnet = __importStar(require("./deardogetestnet"));
exports.desmostestnet = __importStar(require("./desmostestnet"));
exports.doravotatestnet = __importStar(require("./doravotatestnet"));
exports.doravotatestnet2 = __importStar(require("./doravotatestnet2"));
exports.dydxtestnet = __importStar(require("./dydxtestnet"));
exports.elystestnet = __importStar(require("./elystestnet"));
exports.empowertestnet = __importStar(require("./empowertestnet"));
exports.entrypointtestnet = __importStar(require("./entrypointtestnet"));
exports.evmostestnet = __importStar(require("./evmostestnet"));
exports.fetchhubtestnet = __importStar(require("./fetchhubtestnet"));
exports.finschiatestnet = __importStar(require("./finschiatestnet"));
exports.galacticatestnet = __importStar(require("./galacticatestnet"));
exports.gitopiatestnet = __importStar(require("./gitopiatestnet"));
exports.hedgetestnet = __importStar(require("./hedgetestnet"));
exports.humanstestnet = __importStar(require("./humanstestnet"));
exports.hypersigntestnet = __importStar(require("./hypersigntestnet"));
exports.impacthubtestnet = __importStar(require("./impacthubtestnet"));
exports.imversedtestnet = __importStar(require("./imversedtestnet"));
exports.injectivetestnet = __importStar(require("./injectivetestnet"));
exports.jackaltestnet = __importStar(require("./jackaltestnet"));
exports.junotestnet = __importStar(require("./junotestnet"));
exports.kichaintestnet = __importStar(require("./kichaintestnet"));
exports.kujiratestnet = __importStar(require("./kujiratestnet"));
exports.kyvetestnet = __importStar(require("./kyvetestnet"));
exports.lavatestnet = __importStar(require("./lavatestnet"));
exports.lavatestnet1 = __importStar(require("./lavatestnet1"));
exports.likecointestnet = __importStar(require("./likecointestnet"));
exports.lumenxtestnet = __importStar(require("./lumenxtestnet"));
exports.mantrachaintestnet = __importStar(require("./mantrachaintestnet"));
exports.marstestnet = __importStar(require("./marstestnet"));
exports.migalootestnet = __importStar(require("./migalootestnet"));
exports.neutrontestnet = __importStar(require("./neutrontestnet"));
exports.nobletestnet = __importStar(require("./nobletestnet"));
exports.noistestnet = __importStar(require("./noistestnet"));
exports.nolustestnet = __importStar(require("./nolustestnet"));
exports.okp4testnet = __importStar(require("./okp4testnet"));
exports.osmosistestnet = __importStar(require("./osmosistestnet"));
exports.osmosistestnet4 = __importStar(require("./osmosistestnet4"));
exports.permtestnet = __importStar(require("./permtestnet"));
exports.persistencetestnet = __importStar(require("./persistencetestnet"));
exports.persistencetestnet2 = __importStar(require("./persistencetestnet2"));
exports.pryzmtestnet = __importStar(require("./pryzmtestnet"));
exports.quasartestnet = __importStar(require("./quasartestnet"));
exports.quicksilvertestnet = __importStar(require("./quicksilvertestnet"));
exports.qwoyntestnet = __importStar(require("./qwoyntestnet"));
exports.rsprovidertestnet = __importStar(require("./rsprovidertestnet"));
exports.sagatestnet = __importStar(require("./sagatestnet"));
exports.secretnetworktestnet = __importStar(require("./secretnetworktestnet"));
exports.secretnetworktestnet2 = __importStar(require("./secretnetworktestnet2"));
exports.seitestnet = __importStar(require("./seitestnet"));
exports.seitestnet2 = __importStar(require("./seitestnet2"));
exports.selfchaindevnet = __importStar(require("./selfchaindevnet"));
exports.sgetestnet = __importStar(require("./sgetestnet"));
exports.sgetestnet4 = __importStar(require("./sgetestnet4"));
exports.sixtestnet = __importStar(require("./sixtestnet"));
exports.sourcetestnet = __importStar(require("./sourcetestnet"));
exports.stargazetestnet = __importStar(require("./stargazetestnet"));
exports.statesettestnet = __importStar(require("./statesettestnet"));
exports.stridetestnet = __importStar(require("./stridetestnet"));
exports.swisstroniktestnet = __importStar(require("./swisstroniktestnet"));
exports.temporaltestnet = __importStar(require("./temporaltestnet"));
exports.terpnettestnet = __importStar(require("./terpnettestnet"));
exports.terpnettestnet2 = __importStar(require("./terpnettestnet2"));
exports.terra2testnet = __importStar(require("./terra2testnet"));
exports.titantestnet = __importStar(require("./titantestnet"));
exports.ulastestnet = __importStar(require("./ulastestnet"));
exports.uniontestnet = __importStar(require("./uniontestnet"));
exports.upticktestnet = __importStar(require("./upticktestnet"));
exports.vincechaintestnet = __importStar(require("./vincechaintestnet"));
exports.wardenprotocoltestnet = __importStar(require("./wardenprotocoltestnet"));
exports.wavehashtestnet = __importStar(require("./wavehashtestnet"));
exports.xiontestnet = __importStar(require("./xiontestnet"));
exports.zetachaintestnet = __importStar(require("./zetachaintestnet"));
