// chain-registry/noncosmos/named.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ojotestnet = exports.polygontestnet = exports.polkadottestnet = exports.moonbeamtestnet = exports.fantomtestnet = exports.ethereumtestnet = exports.binancesmartchaintestnet = exports.avalanchetestnet = exports.xrpl = exports.tinkernet = exports.sui = exports.stellar = exports.statemine = exports.solana = exports.polygon = exports.polkadot = exports.moonbeam = exports.kusama = exports.filecoin = exports.fantom = exports.ethereum = exports.bitcoin = exports.binancesmartchain = exports.avalanche = exports.arbitrum = exports.aptos = exports.composablepolkadot = exports.picasso = void 0;
exports.picasso = __importStar(require("./picasso"));
exports.composablepolkadot = __importStar(require("./composablepolkadot"));
exports.aptos = __importStar(require("./aptos"));
exports.arbitrum = __importStar(require("./arbitrum"));
exports.avalanche = __importStar(require("./avalanche"));
exports.binancesmartchain = __importStar(require("./binancesmartchain"));
exports.bitcoin = __importStar(require("./bitcoin"));
exports.ethereum = __importStar(require("./ethereum"));
exports.fantom = __importStar(require("./fantom"));
exports.filecoin = __importStar(require("./filecoin"));
exports.kusama = __importStar(require("./kusama"));
exports.moonbeam = __importStar(require("./moonbeam"));
exports.polkadot = __importStar(require("./polkadot"));
exports.polygon = __importStar(require("./polygon"));
exports.solana = __importStar(require("./solana"));
exports.statemine = __importStar(require("./statemine"));
exports.stellar = __importStar(require("./stellar"));
exports.sui = __importStar(require("./sui"));
exports.tinkernet = __importStar(require("./tinkernet"));
exports.xrpl = __importStar(require("./xrpl"));
exports.avalanchetestnet = __importStar(require("./avalanchetestnet"));
exports.binancesmartchaintestnet = __importStar(require("./binancesmartchaintestnet"));
exports.ethereumtestnet = __importStar(require("./ethereumtestnet"));
exports.fantomtestnet = __importStar(require("./fantomtestnet"));
exports.moonbeamtestnet = __importStar(require("./moonbeamtestnet"));
exports.polkadottestnet = __importStar(require("./polkadottestnet"));
exports.polygontestnet = __importStar(require("./polygontestnet"));
exports.ojotestnet = __importStar(require("./ojotestnet"));
